src/traces/funnel/event_data.js

```diff
@@ -0,0 +1,25 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+module.exports = function eventData(out, pt /* , trace, cd, pointNumber */) {
+    // standard cartesian event data
+    out.x = 'xVal' in pt ? pt.xVal : pt.x;
+    out.y = 'yVal' in pt ? pt.yVal : pt.y;
+
+    // for funnel
+    if('percentInitial' in pt) out.percentInitial = pt.percentInitial;
```

Comment:
here `out.percentInitial` should be the raw number (i.e. unformatted).

Reply:
is `percentInitial` between 0-100 or between 0-1?

Reply:
Right now with formatting it is between 0-100. The raw ratios are between 0-1.

Reply:
FYI - in case of `pie` the `pt.percent` contains the `fraction` not the percent.

Reply:
This is what pie does:

https://github.com/plotly/plotly.js/blob/3e3b22dc87d9806ec2d1b82e8d29f572263ce155/src/traces/pie/plot.js#L365-L367

that is:

- `percent` is in fact a _fraction_
- `percentLabel` is a percentage and it is how `hoverinfo` flag `'percent'` shows up in the hover label. It is also how `hovertemplate: '%{percent}'` shows up by default.

For example,

```js
Plotly.newPlot(gd, [{
  type: 'pie',
  labels: ['a', 'b', 'c'],
  values: [1, 2, 3]
}])
```

shows 16.7%, 33.3% and 50% in the hover labels when `'percent'` is part of the `hoverinfo` OR when `hovertemplate: '%{percent}'` (i.e. it uses `percentLabel` behind the scenes). 

But if `hovertemplate: '%{y:.3f}'` we get 0.167, 0.333 and 0.500 in the hover labels (i.e. we format the fraction behind the scenes).